<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>BioHDF: bioh5g_reads</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body onload='searchBox.OnSelectItem(0);'>
<!-- Generated by Doxygen 1.7.4 -->
<script type="text/javascript"><!--
var searchBox = new SearchBox("searchBox", "search",false,'Search');
--></script>
<div id="top">
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">BioHDF&#160;<span id="projectnumber">version 0.3 alpha</span></div>
   <div id="projectbrief">Scalable NGS Data Storage Based on HDF5</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="annotated.html"><span>Data&#160;Structures</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li id="searchli">
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="summary">
<a href="group__bioh5g__reads.html#nested-classes">Data Structures</a> &#124;
<a href="group__bioh5g__reads.html#typedef-members">Typedefs</a> &#124;
<a href="group__bioh5g__reads.html#enum-members">Enumerations</a> &#124;
<a href="group__bioh5g__reads.html#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">bioh5g_reads</div>  </div>
</div>
<div class="contents">

<p>Represents NGS reads (FASTQ, etc.  
<a href="group__bioh5g__reads.html#details">More...</a></p>
<table class="memberdecls">
<tr><td colspan="2"><h2><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a></td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Read data container.  <a href="structbioh5g__read__data.html#details">More...</a><br/></td></tr>
<tr><td colspan="2"><h2><a name="typedef-members"></a>
Typedefs</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gabdf543a50651bcc0b3d161fd685cad8a"></a><!-- doxytag: member="bioh5g_reads::bioh5g_reads" ref="gabdf543a50651bcc0b3d161fd685cad8a" args="" -->
typedef struct _bioh5g_reads *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a></td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">BioHDF reads collection handle. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga2cde2a359b736cff7614cbb073dc4f65"></a><!-- doxytag: member="bioh5g_reads::bioh5g_reads_iterator" ref="ga2cde2a359b736cff7614cbb073dc4f65" args="" -->
typedef struct <br class="typebreak"/>
_bioh5g_reads_iterator *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a></td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">BioHDF reads iterator handle. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga960e1542eb79684f39ce936764e0e323"></a><!-- doxytag: member="bioh5g_reads::bioh5g_reads_properties" ref="ga960e1542eb79684f39ce936764e0e323" args="" -->
typedef struct <br class="typebreak"/>
_bioh5g_reads_properties *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a></td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">BioHDF reads creation properties. <br/></td></tr>
<tr><td colspan="2"><h2><a name="enum-members"></a>
Enumerations</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">enum &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga172fda2ed5ac5409298673248fff64ed">bioh5g_reads_type</a> { <a class="el" href="group__bioh5g__reads.html#gga172fda2ed5ac5409298673248fff64edab0dcef9c3082c344aa4899dfd0ff06da">BASE_SPACE</a>, 
<a class="el" href="group__bioh5g__reads.html#gga172fda2ed5ac5409298673248fff64eda7c89571d49ad76e1497a82aa54dce353">COLOR_SPACE</a>
 }</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Describes the data "space" of the reads (base, color, etc.)  <a href="group__bioh5g__reads.html#ga172fda2ed5ac5409298673248fff64ed">More...</a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">enum &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">bioh5g_reads_format</a> { <a class="el" href="group__bioh5g__reads.html#ggaec86a71ea29336cee4b71be880c72e4badd065612c72265ce76751419fbb47293">FASTQ_FORMAT</a>, 
<a class="el" href="group__bioh5g__reads.html#ggaec86a71ea29336cee4b71be880c72e4ba23c1dda1284298405ae9f85b868c935d">FASTA_FORMAT</a>
 }</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">The text file format for read I/O.  <a href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">More...</a><br/></td></tr>
<tr><td colspan="2"><h2><a name="func-members"></a>
Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gacf418fa4b9b22296725e8a66e2e9172f">BIOH5Gcheck_reads_presence</a> (<a class="el" href="group__biohdf__file.html#gaf8e0d1bda4736ead6bdb756bed422df6">biohdf_file</a> file, const char *path, int *presence)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Test if a reads collection exists.  <a href="group__bioh5g__reads.html#gacf418fa4b9b22296725e8a66e2e9172f"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga38faf9bdd3ddea6b56fc89f6fa1f5749">BIOH5Gcreate_reads_collection</a> (<a class="el" href="group__biohdf__file.html#gaf8e0d1bda4736ead6bdb756bed422df6">biohdf_file</a> file, <a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> properties, const char *path, <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> *reads)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Create (and open) a new reads collection.  <a href="group__bioh5g__reads.html#ga38faf9bdd3ddea6b56fc89f6fa1f5749"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gaaaacf04b0624fd11669f61760014cf82">BIOH5Gopen_reads_collection</a> (<a class="el" href="group__biohdf__file.html#gaf8e0d1bda4736ead6bdb756bed422df6">biohdf_file</a> file, const char *path, <a class="el" href="group__biohdf__api.html#ga898d970f64f8016f805a7c8cea272036">biohdf_open_mode</a> mode, <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> *reads)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Open an existing reads collection.  <a href="group__bioh5g__reads.html#gaaaacf04b0624fd11669f61760014cf82"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga6e60e3c27e915185879b797c05273e56">BIOH5Gclose_reads_collection</a> (<a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> *reads)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Close an open reads collection.  <a href="group__bioh5g__reads.html#ga6e60e3c27e915185879b797c05273e56"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga568f7c66d66124adac382afa586f71e6">BIOH5Gget_reads_count</a> (const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> reads, int64_t *count)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of stored reads in a collection.  <a href="group__bioh5g__reads.html#ga568f7c66d66124adac382afa586f71e6"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga0a6041dd7eb58ea2ee9be0a5733d4698">BIOH5Gcreate_reads_iterator</a> (const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> reads, <a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a> *iter)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Create an iterator for a reads collection.  <a href="group__bioh5g__reads.html#ga0a6041dd7eb58ea2ee9be0a5733d4698"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga9ac8c8dd290e87cc4b6ef4a7a2cfc441">BIOH5Gdestroy_reads_iterator</a> (<a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a> *iter)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Destroy an iterator for a reads collection.  <a href="group__bioh5g__reads.html#ga9ac8c8dd290e87cc4b6ef4a7a2cfc441"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga28084b2a7928b11fa8663c6e939ee3aa">BIOH5Gadd_read</a> (const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> reads, const <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Add a read to a collection.  <a href="group__bioh5g__reads.html#ga28084b2a7928b11fa8663c6e939ee3aa"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga893641193ea4ce6ecc9db8a5fa932d35">BIOH5Gget_index_of_last_added_read</a> (const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> reads, int64_t *index)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the index of the last read that was added.  <a href="group__bioh5g__reads.html#ga893641193ea4ce6ecc9db8a5fa932d35"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga6a90a823912aabfdf8d464ba90236f18">BIOH5Gget_next_read</a> (<a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a> iter, int64_t *index, <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **data)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the next read from a reads collection.  <a href="group__bioh5g__reads.html#ga6a90a823912aabfdf8d464ba90236f18"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gaeeca0e7227e5b71138d94c0fc337e44b">BIOH5Gget_read</a> (const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> reads, int64_t index, <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **data)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Given a read index, get the read from a reads collection.  <a href="group__bioh5g__reads.html#gaeeca0e7227e5b71138d94c0fc337e44b"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga22fa588815bd00b216ecfe32c4090a42">BIOH5Gfree_read</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **data)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Free read data that has been obtained from the library.  <a href="group__bioh5g__reads.html#ga22fa588815bd00b216ecfe32c4090a42"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga01e7446d72d11404cefeec18f8f9e5b2">BIOH5Gcreate_read_string</a> (const <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *read, <a class="el" href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">bioh5g_reads_format</a> format, char **read_string)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Create a read string in a given format (FASTQ, etc.)  <a href="group__bioh5g__reads.html#ga01e7446d72d11404cefeec18f8f9e5b2"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gac456299be985ee5cad9518314588bcea">BIOH5Gwrite_read_to_stream</a> (const <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *read, <a class="el" href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">bioh5g_reads_format</a> format, FILE *stream)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Output a read to a stream in a given format.  <a href="group__bioh5g__reads.html#gac456299be985ee5cad9518314588bcea"></a><br/></td></tr>
<tr><td colspan="2"><h2><a name="member-group"></a>
Functions: Data Accessors</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gad6bf4603202002475490a8b932bb0c94"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gcreate_read_data" ref="gad6bf4603202002475490a8b932bb0c94" args="(bioh5g_read_data **data)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gad6bf4603202002475490a8b932bb0c94">BIOH5Gcreate_read_data</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **data)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Create a read data container. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gadfc84d6fc17fa522c9e3d81e2261cbda"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gget_read_identifier" ref="gadfc84d6fc17fa522c9e3d81e2261cbda" args="(bioh5g_read_data *data, char **identifier)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gadfc84d6fc17fa522c9e3d81e2261cbda">BIOH5Gget_read_identifier</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data, char **identifier)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the read identifier. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga86d85188e2b30ae37b6630da46bfe9bd"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_read_identifier" ref="ga86d85188e2b30ae37b6630da46bfe9bd" args="(bioh5g_read_data *data, char *identifier)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga86d85188e2b30ae37b6630da46bfe9bd">BIOH5Gset_read_identifier</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data, char *identifier)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the read identifier. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gaa120766c3bf2fac7baedc9a27bb452e0"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gget_read_sequence" ref="gaa120766c3bf2fac7baedc9a27bb452e0" args="(bioh5g_read_data *data, char **sequence)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gaa120766c3bf2fac7baedc9a27bb452e0">BIOH5Gget_read_sequence</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data, char **sequence)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the read sequence. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga31e58f9b1bcd73a0fdeeb89d07080565"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_read_sequence" ref="ga31e58f9b1bcd73a0fdeeb89d07080565" args="(bioh5g_read_data *data, char *sequence)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga31e58f9b1bcd73a0fdeeb89d07080565">BIOH5Gset_read_sequence</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data, char *sequence)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the read sequence. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga7749c53c6a37c9cb9377746c154a78fb"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gget_read_quality_values" ref="ga7749c53c6a37c9cb9377746c154a78fb" args="(bioh5g_read_data *data, char **quality_values)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga7749c53c6a37c9cb9377746c154a78fb">BIOH5Gget_read_quality_values</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data, char **quality_values)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the read quality values. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga2c08761e715aece1c23f703515e560ae"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_read_quality_values" ref="ga2c08761e715aece1c23f703515e560ae" args="(bioh5g_read_data *data, char *quality_values)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga2c08761e715aece1c23f703515e560ae">BIOH5Gset_read_quality_values</a> (<a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *data, char *quality_values)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the read quality values. <br/></td></tr>
<tr><td colspan="2"><h2><a name="member-group"></a>
Functions: Collection Creation Properties</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga2a3db269f989389e8a9c8fa6bb9e99d1"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gcreate_reads_properties" ref="ga2a3db269f989389e8a9c8fa6bb9e99d1" args="(bioh5g_reads_properties *props)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga2a3db269f989389e8a9c8fa6bb9e99d1">BIOH5Gcreate_reads_properties</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> *props)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Create a reads properties container. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gab9bae3ee8210b430dc48f6a1d1a9786e"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gdestroy_reads_properties" ref="gab9bae3ee8210b430dc48f6a1d1a9786e" args="(bioh5g_reads_properties *props)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gab9bae3ee8210b430dc48f6a1d1a9786e">BIOH5Gdestroy_reads_properties</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> *props)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Destroy a reads properties container. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gae03c1260693d7dc583f07aaceebf9a75"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_reads_type" ref="gae03c1260693d7dc583f07aaceebf9a75" args="(bioh5g_reads_properties props, bioh5g_reads_type reads_type)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gae03c1260693d7dc583f07aaceebf9a75">BIOH5Gset_reads_properties_reads_type</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, <a class="el" href="group__bioh5g__reads.html#ga172fda2ed5ac5409298673248fff64ed">bioh5g_reads_type</a> reads_type)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties reads type. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga1d8c56e2efbef6cc24198926fb1be46c"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_chunk_size" ref="ga1d8c56e2efbef6cc24198926fb1be46c" args="(bioh5g_reads_properties props, int64_t chunk_size)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga1d8c56e2efbef6cc24198926fb1be46c">BIOH5Gset_reads_properties_chunk_size</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, int64_t chunk_size)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties chunk size. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gae90fc44c3618ad88cbd10f335182026d"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_compression_level" ref="gae90fc44c3618ad88cbd10f335182026d" args="(bioh5g_reads_properties props, compression_level level)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gae90fc44c3618ad88cbd10f335182026d">BIOH5Gset_reads_properties_compression_level</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, <a class="el" href="group__biohdf__api.html#ga3e3fea2fdf249a0f887b02d08e454ec2">compression_level</a> level)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties compression level. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="gaf158c540ae914b9c2a71428925fec87b"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_sequences_scheme" ref="gaf158c540ae914b9c2a71428925fec87b" args="(bioh5g_reads_properties props, biohdf_string_storage_scheme scheme)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#gaf158c540ae914b9c2a71428925fec87b">BIOH5Gset_reads_properties_sequences_scheme</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, <a class="el" href="group__biohdf__api.html#ga56ee55722c96e73748569b7debb1e8ba">biohdf_string_storage_scheme</a> scheme)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties sequences storage scheme. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga60323003862b12cbf94b333f8133532e"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_identifiers_scheme" ref="ga60323003862b12cbf94b333f8133532e" args="(bioh5g_reads_properties props, biohdf_string_storage_scheme scheme)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga60323003862b12cbf94b333f8133532e">BIOH5Gset_reads_properties_identifiers_scheme</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, <a class="el" href="group__biohdf__api.html#ga56ee55722c96e73748569b7debb1e8ba">biohdf_string_storage_scheme</a> scheme)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties identifiers storage scheme. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga9bede3d1d798116ddad71a62258a50c4"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_sequences_length" ref="ga9bede3d1d798116ddad71a62258a50c4" args="(bioh5g_reads_properties props, size_t length)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga9bede3d1d798116ddad71a62258a50c4">BIOH5Gset_reads_properties_sequences_length</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, size_t length)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties sequences length. <br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ga2650df6b295d8423bbdcfcd7288a15b1"></a><!-- doxytag: member="bioh5g_reads::BIOH5Gset_reads_properties_identifiers_length" ref="ga2650df6b295d8423bbdcfcd7288a15b1" args="(bioh5g_reads_properties props, size_t length)" -->
BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__bioh5g__reads.html#ga2650df6b295d8423bbdcfcd7288a15b1">BIOH5Gset_reads_properties_identifiers_length</a> (<a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a> props, size_t length)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Set reads properties identifiers length. <br/></td></tr>
</table>
<hr/><a name="details" id="details"></a><h2>Detailed Description</h2>
<p>Represents NGS reads (FASTQ, etc. </p>
<p>entries). </p>
<hr/><h2>Enumeration Type Documentation</h2>
<a class="anchor" id="gaec86a71ea29336cee4b71be880c72e4b"></a><!-- doxytag: member="bioh5g_reads.h::bioh5g_reads_format" ref="gaec86a71ea29336cee4b71be880c72e4b" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">enum <a class="el" href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">bioh5g_reads_format</a></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>The text file format for read I/O. </p>
<p>BioHDF stores the FASTQ quality values verbatim and does no attempt to convert or normalize them between quality score schemes. </p>
<dl><dt><b>Enumerator: </b></dt><dd><table border="0" cellspacing="2" cellpadding="0">
<tr><td valign="top"><em><a class="anchor" id="ggaec86a71ea29336cee4b71be880c72e4badd065612c72265ce76751419fbb47293"></a><!-- doxytag: member="FASTQ_FORMAT" ref="ggaec86a71ea29336cee4b71be880c72e4badd065612c72265ce76751419fbb47293" args="" -->FASTQ_FORMAT</em>&nbsp;</td><td>
<p>FASTQ format. </p>
</td></tr>
<tr><td valign="top"><em><a class="anchor" id="ggaec86a71ea29336cee4b71be880c72e4ba23c1dda1284298405ae9f85b868c935d"></a><!-- doxytag: member="FASTA_FORMAT" ref="ggaec86a71ea29336cee4b71be880c72e4ba23c1dda1284298405ae9f85b868c935d" args="" -->FASTA_FORMAT</em>&nbsp;</td><td>
<p>FASTA (or CSFASTA) format. </p>
</td></tr>
</table>
</dd>
</dl>

</div>
</div>
<a class="anchor" id="ga172fda2ed5ac5409298673248fff64ed"></a><!-- doxytag: member="bioh5g_reads.h::bioh5g_reads_type" ref="ga172fda2ed5ac5409298673248fff64ed" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">enum <a class="el" href="group__bioh5g__reads.html#ga172fda2ed5ac5409298673248fff64ed">bioh5g_reads_type</a></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Describes the data "space" of the reads (base, color, etc.) </p>
<p>Reads are stored verbatim and not converted between spaces. </p>
<dl><dt><b>Enumerator: </b></dt><dd><table border="0" cellspacing="2" cellpadding="0">
<tr><td valign="top"><em><a class="anchor" id="gga172fda2ed5ac5409298673248fff64edab0dcef9c3082c344aa4899dfd0ff06da"></a><!-- doxytag: member="BASE_SPACE" ref="gga172fda2ed5ac5409298673248fff64edab0dcef9c3082c344aa4899dfd0ff06da" args="" -->BASE_SPACE</em>&nbsp;</td><td>
<p>Base/sequence space reads. </p>
</td></tr>
<tr><td valign="top"><em><a class="anchor" id="gga172fda2ed5ac5409298673248fff64eda7c89571d49ad76e1497a82aa54dce353"></a><!-- doxytag: member="COLOR_SPACE" ref="gga172fda2ed5ac5409298673248fff64eda7c89571d49ad76e1497a82aa54dce353" args="" -->COLOR_SPACE</em>&nbsp;</td><td>
<p>Color space (SOLiD) reads. </p>
</td></tr>
</table>
</dd>
</dl>

</div>
</div>
<hr/><h2>Function Documentation</h2>
<a class="anchor" id="ga28084b2a7928b11fa8663c6e939ee3aa"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gadd_read" ref="ga28084b2a7928b11fa8663c6e939ee3aa" args="(const bioh5g_reads reads, const bioh5g_read_data *data)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gadd_read </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a>&#160;</td>
          <td class="paramname"><em>reads</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *&#160;</td>
          <td class="paramname"><em>data</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Add a read to a collection. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramname">reads</td><td>A BioHDF reads handle </td></tr>
    <tr><td class="paramname">data</td><td>A BioHDF read</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="gacf418fa4b9b22296725e8a66e2e9172f"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gcheck_reads_presence" ref="gacf418fa4b9b22296725e8a66e2e9172f" args="(biohdf_file file, const char *path, int *presence)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gcheck_reads_presence </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__biohdf__file.html#gaf8e0d1bda4736ead6bdb756bed422df6">biohdf_file</a>&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>path</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int *&#160;</td>
          <td class="paramname"><em>presence</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Test if a reads collection exists. </p>
<p>This function will return TRUE if a collection of the same type exists at the named location. If any other HDF5 or BioHDF object with that same name exists, TRUE will be returned as well as an error code, the assumption being that Bad Code(tm) that does not check return values will be more likely to attempt to open code (and fail), rather than create things (which my partially succeed, making a mess).</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">file</td><td>A BioHDF file handle </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">path</td><td>The BioHDF path to the collection </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">presence</td><td>TRUE if the collection exists, FALSE if it does not.</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga6e60e3c27e915185879b797c05273e56"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gclose_reads_collection" ref="ga6e60e3c27e915185879b797c05273e56" args="(bioh5g_reads *reads)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gclose_reads_collection </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> *&#160;</td>
          <td class="paramname"><em>reads</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Close an open reads collection. </p>
<p>This function will set the collection handle to NULL after freeing it.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir">[in,out]</td><td class="paramname">reads</td><td>A BioHDF reads handle</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga01e7446d72d11404cefeec18f8f9e5b2"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gcreate_read_string" ref="ga01e7446d72d11404cefeec18f8f9e5b2" args="(const bioh5g_read_data *read, bioh5g_reads_format format, char **read_string)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gcreate_read_string </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *&#160;</td>
          <td class="paramname"><em>read</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">bioh5g_reads_format</a>&#160;</td>
          <td class="paramname"><em>format</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char **&#160;</td>
          <td class="paramname"><em>read_string</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Create a read string in a given format (FASTQ, etc.) </p>
<p>The read sequence and identifier must not be NULL. If the format is FASTQ, the quality values must not be NULL. The sequence and quality values must also have the same length for FASTQ output.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">read</td><td>The BioHDF read </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">format</td><td>The format for the output string </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">read_string</td><td>The output string</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga38faf9bdd3ddea6b56fc89f6fa1f5749"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gcreate_reads_collection" ref="ga38faf9bdd3ddea6b56fc89f6fa1f5749" args="(biohdf_file file, bioh5g_reads_properties properties, const char *path, bioh5g_reads *reads)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gcreate_reads_collection </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__biohdf__file.html#gaf8e0d1bda4736ead6bdb756bed422df6">biohdf_file</a>&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#ga960e1542eb79684f39ce936764e0e323">bioh5g_reads_properties</a>&#160;</td>
          <td class="paramname"><em>properties</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>path</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> *&#160;</td>
          <td class="paramname"><em>reads</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Create (and open) a new reads collection. </p>
<p>The collection handle returned by this function will be ready to accept I/O.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">file</td><td>A BioHDF file handle </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">properties</td><td>Collection creation properties </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">path</td><td>The BioHDF path to the collection </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">reads</td><td>A BioHDF reads handle</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga0a6041dd7eb58ea2ee9be0a5733d4698"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gcreate_reads_iterator" ref="ga0a6041dd7eb58ea2ee9be0a5733d4698" args="(const bioh5g_reads reads, bioh5g_reads_iterator *iter)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gcreate_reads_iterator </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a>&#160;</td>
          <td class="paramname"><em>reads</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a> *&#160;</td>
          <td class="paramname"><em>iter</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Create an iterator for a reads collection. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">reads</td><td>A BioHDF reads handle </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">iter</td><td>An iterator for a reads collection</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga9ac8c8dd290e87cc4b6ef4a7a2cfc441"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gdestroy_reads_iterator" ref="ga9ac8c8dd290e87cc4b6ef4a7a2cfc441" args="(bioh5g_reads_iterator *iter)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gdestroy_reads_iterator </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a> *&#160;</td>
          <td class="paramname"><em>iter</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Destroy an iterator for a reads collection. </p>
<p>The iterator is set to NULL as a part of deletion.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir">[in,out]</td><td class="paramname">iter</td><td>An iterator for a reads collection</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

</div>
</div>
<a class="anchor" id="ga22fa588815bd00b216ecfe32c4090a42"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gfree_read" ref="ga22fa588815bd00b216ecfe32c4090a42" args="(bioh5g_read_data **data)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gfree_read </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **&#160;</td>
          <td class="paramname"><em>data</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Free read data that has been obtained from the library. </p>
<p>The data is set to NULL as a part of deletion.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir">[in,out]</td><td class="paramname">data</td><td>The BioHDF read</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

</div>
</div>
<a class="anchor" id="ga893641193ea4ce6ecc9db8a5fa932d35"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gget_index_of_last_added_read" ref="ga893641193ea4ce6ecc9db8a5fa932d35" args="(const bioh5g_reads reads, int64_t *index)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gget_index_of_last_added_read </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a>&#160;</td>
          <td class="paramname"><em>reads</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int64_t *&#160;</td>
          <td class="paramname"><em>index</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Get the index of the last read that was added. </p>
<p>Useful for adding SAM lines where a link to a given read must be created.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">reads</td><td>A BioHDF reads handle </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">index</td><td>The index of the last read that was added</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga6a90a823912aabfdf8d464ba90236f18"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gget_next_read" ref="ga6a90a823912aabfdf8d464ba90236f18" args="(bioh5g_reads_iterator iter, int64_t *index, bioh5g_read_data **data)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gget_next_read </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#ga2cde2a359b736cff7614cbb073dc4f65">bioh5g_reads_iterator</a>&#160;</td>
          <td class="paramname"><em>iter</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int64_t *&#160;</td>
          <td class="paramname"><em>index</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **&#160;</td>
          <td class="paramname"><em>data</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Get the next read from a reads collection. </p>
<p>When the iterator has finished traversing the collection, data will be NULL, index will be -1 and the return value will be BIOHDF_NO_ERROR.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">iter</td><td>An iterator for a reads collection </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">index</td><td>The index of this read </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">data</td><td>The BioHDF read</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="gaeeca0e7227e5b71138d94c0fc337e44b"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gget_read" ref="gaeeca0e7227e5b71138d94c0fc337e44b" args="(const bioh5g_reads reads, int64_t index, bioh5g_read_data **data)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gget_read </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a>&#160;</td>
          <td class="paramname"><em>reads</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int64_t&#160;</td>
          <td class="paramname"><em>index</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> **&#160;</td>
          <td class="paramname"><em>data</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Given a read index, get the read from a reads collection. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">reads</td><td>A BioHDF reads handle </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">index</td><td>The index of this read </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">data</td><td>The BioHDF read</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="ga568f7c66d66124adac382afa586f71e6"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gget_reads_count" ref="ga568f7c66d66124adac382afa586f71e6" args="(const bioh5g_reads reads, int64_t *count)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gget_reads_count </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a>&#160;</td>
          <td class="paramname"><em>reads</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int64_t *&#160;</td>
          <td class="paramname"><em>count</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Get the number of stored reads in a collection. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">reads</td><td>A BioHDF reads handle </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">count</td><td>The number of reads in the collection</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="gaaaacf04b0624fd11669f61760014cf82"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gopen_reads_collection" ref="gaaaacf04b0624fd11669f61760014cf82" args="(biohdf_file file, const char *path, biohdf_open_mode mode, bioh5g_reads *reads)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gopen_reads_collection </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="group__biohdf__file.html#gaf8e0d1bda4736ead6bdb756bed422df6">biohdf_file</a>&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>path</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__biohdf__api.html#ga898d970f64f8016f805a7c8cea272036">biohdf_open_mode</a>&#160;</td>
          <td class="paramname"><em>mode</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#gabdf543a50651bcc0b3d161fd685cad8a">bioh5g_reads</a> *&#160;</td>
          <td class="paramname"><em>reads</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Open an existing reads collection. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">file</td><td>A BioHDF file handle </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">path</td><td>The BioHDF path to the collection </td></tr>
    <tr><td class="paramdir"></td><td class="paramname">mode</td><td>The access mode (read-only | read-write) </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">reads</td><td>A BioHDF reads handle</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
<a class="anchor" id="gac456299be985ee5cad9518314588bcea"></a><!-- doxytag: member="bioh5g_reads.h::BIOH5Gwrite_read_to_stream" ref="gac456299be985ee5cad9518314588bcea" args="(const bioh5g_read_data *read, bioh5g_reads_format format, FILE *stream)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">BIOHDF_API <a class="el" href="group__biohdf__error.html#ga937fa9c5ae2aeda743ed50b74e340480">biohdf_error</a> BIOH5Gwrite_read_to_stream </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="structbioh5g__read__data.html">bioh5g_read_data</a> *&#160;</td>
          <td class="paramname"><em>read</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__bioh5g__reads.html#gaec86a71ea29336cee4b71be880c72e4b">bioh5g_reads_format</a>&#160;</td>
          <td class="paramname"><em>format</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">FILE *&#160;</td>
          <td class="paramname"><em>stream</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Output a read to a stream in a given format. </p>
<p>This saves you from having to create temp strings that will just be dumped to a stream.</p>
<p>The read sequence and identifier must not be NULL. If the format is FASTQ, the quality values must not be NULL. The sequence and quality values must also have the same length for FASTQ output.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramname">read</td><td>The BioHDF read </td></tr>
    <tr><td class="paramname">format</td><td>The format for the output string </td></tr>
    <tr><td class="paramname">stream</td><td>The output stream (can be STDOUT)</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>BIOHDF_NO_ERROR on success, a BioHDF error code on failure </dd></dl>

<p><p>CHECK*PARAMETERS</p>
<p>CODE</p>
<p>SUCCESS</p>
<p>FAILURE </p>
</p>

</div>
</div>
</div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Data Structures</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Variables</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<hr class="footer"/><address class="footer"><small>Generated on Fri Dec 23 2011 11:59:36 for BioHDF by&#160;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.4 </small></address>
</body>
</html>
